src/ast.rs

```diff
@@ -355,11 +372,22 @@ impl<'i> HvmcParser<'i> {
 
   /// See `ops.rs` for the available operators.
   fn parse_op(&mut self) -> Result<Op, String> {
-    let op = self.take_while(|c| "ui0123456789.+-=*/%<>|&^!?$".contains(c));
+    let op = self.take_while(|c| c.is_alphanumeric() || ".+-=*/%<>|&^!?$".contains(c));
     op.parse().map_err(|_| format!("Unknown operator: {op:?}"))
   }
 }
 
+/// Parses an unsigned integer with an optional radix prefix.
+fn parse_int(input: &str) -> Result<u64, String> {
```

Comment:
I believe TSPL has a `parse_u64` method you can use

Reply:
I'm deferring parsing here until I can confirm that the alphanumerics I've read are indeed an integer.